### PR TITLE
[IOS-597] implement pushed search header on new group controller

### PR DESCRIPTION
**Pull request by marijnschilling**
I aded the pushed search header to group chat 🎉 

But while doing that i realised a whole bunch of things can go now. We used to use the `ProfileviewController` for the favorites, for creating a new chat, for creating a new group and for editing group members. Now we only use it for the last two cases. So i cleaned up that whole controller and removed all unnecessary stuff as well. 
- [ ] removed the profileHeaderView that adds a searchBar to it in <iOS11 (so basically removed a whole bunch of `if iOS11 is available lalalala`)
- [ ] I added the `ProfilesAddedToGroupHeader` as a normal view (no more tableViewHeader bs)
- [ ] Removed the custom `BrowseSearchResultsTableView` and added this as _the_ tableView to the profile view, since the favorites is gone anyway only search table view is there.
- [ ] removed the last remnants of favorites in localized strings and other things that were hanging around.

Create new group:
new group

Add to group:
add to group

### CODE REVIEW

File: Toshi/Controllers/Profiles/ProfilesViewController.swift [hunk 1]
```diff
@@ -18,14 +18,11 @@ import UIKit
 // MARK: - Profiles View Controller Type
 
 public enum ProfilesViewControllerType {
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Controllers/Profiles/ProfilesViewController.swift]: for me it is kind of "picker", user picker .. the purpose of which is to select a few users from some list.. but i do not mind it to stay like this 🤷‍♂️

File: Toshi/Controllers/Chats/FindUsersBotsGroupsViewController.swift [hunk 2]
```diff
@@ -93,6 +92,10 @@ final class FindUsersBotsGroupsViewController: UIViewController {
 }
 
 extension FindUsersBotsGroupsViewController: PushedSearchHeaderDelegate {
+    func searchHeaderDidEnterSearchTerm(_ searchText: String, _ headerView: PushedSearchHeaderView) {
+
```
Comment (yuliaveres) [inline Toshi/Controllers/Chats/FindUsersBotsGroupsViewController.swift]: empty ?

File: Toshi/Controllers/Chats/NewChatViewController.swift [hunk 3]
```diff
@@ -115,6 +115,10 @@ final class NewChatViewController: UIViewController {
 }
 
 extension NewChatViewController: PushedSearchHeaderDelegate {
+    func searchHeaderDidEnterSearchTerm(_ searchText: String, _ headerView: PushedSearchHeaderView) {
+
```
Comment (yuliaveres) [inline Toshi/Controllers/Chats/NewChatViewController.swift]: empty or just leftover ?

File: Toshi/Views/PushedSearchHeaderView.swift [hunk 4]
```diff
@@ -16,18 +16,26 @@
...
     func searchHeaderWillBeginEditing(_ headerView: PushedSearchHeaderView)
     func searchHeaderWillEndEditing(_ headerView: PushedSearchHeaderView)
 
-    func searchHeaderDidReceiveCancelEvent(_ headerView: PushedSearchHeaderView)
+    func searchHeaderDidReceiveRightButtonEvent(_ headerView: PushedSearchHeaderView)
     func searchHeaderViewDidReceiveBackEvent(_ headerView: PushedSearchHeaderView)
 }
 
+enum PushedSearchHeaderViewState: Int {
+    case select, normal
```
[the remark this answers is not in the document]
Comment (yuliaveres) [reply Toshi/Views/PushedSearchHeaderView.swift]: does the title of the right button depend on this only ?
if so - lets make it as accessible property so controller can say `pushedHeader.rightButtonTitle = "whatever"` so it is much more flexible, won't need to add any more cases if a new title is needed, and won't need to think on naming of these cases

File: Toshi/Views/PushedSearchHeaderView.swift [hunk 5]
```diff
@@ -138,14 +168,16 @@ final class PushedSearchHeaderView: UIView {
...
+        rightButton.isHidden = true
     }
 
     private func adjustToSearching(isSearching: Bool) {
         layoutIfNeeded()
-        cancelButton.isHidden = !isSearching
-        backButton.isHidden = isSearching
+        rightButton.isHidden = !isSearching
 
+        guard state == .normal else { return }
```
Comment (yuliaveres) [inline Toshi/Views/PushedSearchHeaderView.swift]: also can go to property `hidesRightButtonOnSearch` which by default might be YES / true

File: Toshi/Views/PushedSearchHeaderView.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (marijnschilling) [reply Toshi/Views/PushedSearchHeaderView.swift]: No it also decides wether the button should be enabled

File: Toshi/Controllers/Chats/FindUsersBotsGroupsViewController.swift (hunk 2 above)
Comment (marijnschilling) [reply Toshi/Controllers/Chats/FindUsersBotsGroupsViewController.swift]: We will need to implement this later on, i can place a comment in it!` //TODO:`

File: Toshi/Views/PushedSearchHeaderView.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (marijnschilling) [reply Toshi/Views/PushedSearchHeaderView.swift]: But yeah i think it's a good point. If we would even need a third isolated state to use this view we can add an enum. But for now just vars is easier to make it more flexible ✌️

File: Toshi/Controllers/Profiles/ProfilesViewController.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (designatednerd) [reply Toshi/Controllers/Profiles/ProfilesViewController.swift]: I'm good with `GroupParticipantsViewController` but I think we can also postpone dealing with the naming for a bit

File: Toshi/Views/PushedSearchHeaderView.swift [hunk 6]
```diff
@@ -138,14 +156,16 @@ final class PushedSearchHeaderView: UIView {
...
+        rightButton.isHidden = true
     }
 
     private func adjustToSearching(isSearching: Bool) {
         layoutIfNeeded()
-        cancelButton.isHidden = !isSearching
-        backButton.isHidden = isSearching
+        rightButton.isHidden = !isSearching
+
+        guard hidesRightButtonOnSearch else { return }
```
Comment (designatednerd) [inline Toshi/Views/PushedSearchHeaderView.swift]: should this `guard` be above where `rightButton.isHidden` is set?

Comment (marijnschilling) [reply Toshi/Views/PushedSearchHeaderView.swift]: oops, good catch! the place is right, the name was wrong. We're hiding the backButton on search in some cases, not the right button.

**Pull request merged**